### DISCUSSION TITLE
Easily identifying internal systems in events and hunting

### DISCUSSION THREAD

**Discussion by Kleinco** [Security-Onion-Solutions/securityonion#6743] [Unsupported Versions]
Hey everyone,

I'm wondering if there are easier ways to identify key internal systems when hunting and reviewing events, rather than having to cross-reference internal IP addresses against DHCP logs. 

Can someone please offer any advice or point me in the right direction (other than allocating static internal addresses). My initial thinking is to include MAC addresses in events and configure a lookup of MAC address to internal host. 

I've reviewed the documentation but am not sure how best to proceed. Is this the best approach? Does it require a custom Elastic parser? Is it better configured with a Salt pillar? And where should a MAC-to-hostname lookup be performed?

(I appreciate that external hosts will only show the internal MAC addresses of the gateway).

Many thanks in advance for anyone who can help :)

**Comment by weslambert** [comment]
Hi Nick,

Here are a few options (off the top of my head -- there are likely others as well):

1) Include MAC addresses in the logs (through Zeek mac-logging or DHCP logs), populate an enrich index with the MAC/hostname (from external source), then use the enrich processor in an ingest node pipeline to make the comparison. This means all of this is accomplished at processing/index time and a new field would be added that includes this data. Keep in mind, you have to manage and re-create/update the enrich indices at a scheduled interval as your inventory is updated.

    See: https://www.elastic.co/guide/en/elasticsearch/reference/current/enrich-processor.html

2) Include MAC addresses in the logs (through Zeek mac-logging or DHCP logs), then configure an ad-hoc lookup action from SOC Hunt/Alerts. This would leverage the field containing the value of the MAC address and perform a lookup to an external service you control. This would not modify the data, and would need to be initiated per record.

    See: https://docs.securityonion.net/en/latest/alerts.html?highlight=actions#actions

3) Not recommended - Include MAC addresses in the logs (through Zeek mac-logging or DHCP logs), then modify a Logstash pipeline to use the translate filter in the config, leveraging an externally-provided file/mapping.

    See: https://www.elastic.co/guide/en/logstash/current/plugins-filters-translate.html

Hopefully, that helps to get you going in the right direction.

Thanks,
Wes

**Reply by Kleinco** [reply]
Hey Wes, thanks so much for taking the time to offer these options - I really appreciate it.

I'll work through these options and see what's best for my setup.

Best regards,

Nick.
